src/soc/intel/common/block/oc_wdt/acpi.c

```diff
@@ -0,0 +1,156 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include <acpi/acpi.h>
+#include <console/console.h>
+#include <intelblocks/oc_wdt.h>
+
+#define OC_WDT_MIN_VALUE	0
+#define OC_WDT_MAX_VALUE	1023
+#define OC_WDT_PERIOD_MS	1000
+
+static bool fill_wdat_timeout_entry(acpi_wdat_entry_t *entry)
+{
+	memset((void *)entry, 0, sizeof(acpi_wdat_entry_t));
+
+	entry->action = ACPI_WDAT_SET_COUNTDOWN;
+	entry->instruction = ACPI_WDAT_WRITE_COUNTDOWN | ACPI_WDAT_PRESERVE_REGISTER;
```

Comment:
Bits 24 and 25 are write-1-clear, so they shouldn't be preserved. If OS writes 1 to these, they will be cleared.

Reply:
https://github.com/Dasharo/coreboot/pull/472/commits/f99cf95497da3151fe21dc6806c23af9b32e5129

Reply:
I knew that happens, but I wasn't sure if it was OK to access the register at lower access widths. Will test with your suggestions and report back

Reply:
As long as your accesses are aligned i.e. word access on word boundary, it should be fine. Byte access has no resctrictions.